sdk/nodejs/cmd/pulumi-language-nodejs/main.go

```diff
@@ -569,7 +573,13 @@ func (host *nodeLanguageHost) Run(ctx context.Context, req *pulumirpc.RunRequest
 		runPath = defaultRunPath
 	}
 
-	runPath, err = locateModule(ctx, runPath, req.Info.ProgramDirectory, nodeBin)
+	// If we're forcing tsc the program directory for running is actually ./bin
+	programDirectory := req.Info.ProgramDirectory
+	if host.forceTsc {
+		req.Info.ProgramDirectory = filepath.Join(programDirectory, "bin")
```

Comment:
Does it work if we set `main: bin/index.js` in package.json instead of this?

Reply:
That was my first thought but it upsets the engine because the "bin" directory doesn't exist to start with.